src/cider/tests/hashtable/CiderHashTableTest.cpp

```diff
@@ -60,6 +65,43 @@ int random(int low, int high) {
   return dist(gen);
 }
 
+cider_hashtable::HashTableRegistrar<
+    cider_hashtable::BaseHashTable<
+        int,
+        int,
+        MurmurHash,
+        Equal,
+        void,
+        std::allocator<std::pair<cider_hashtable::table_key<int>, int>>>,
+    cider_hashtable::LinearProbeHashTable<int, int, MurmurHash, Equal>>
+    linear_hashtable("linear_int");
+cider_hashtable::HashTableRegistrar<
+    cider_hashtable::BaseHashTable<
+        int,
+        std::pair<cider::exec::nextgen::context::Batch*, int>,
+        MurmurHash,
+        Equal,
+        void,
+        std::allocator<std::pair<cider_hashtable::table_key<int>,
+                                 std::pair<cider::exec::nextgen::context::Batch*, int>>>>,
+    cider_hashtable::LinearProbeHashTable<
+        int,
+        std::pair<cider::exec::nextgen::context::Batch*, int>,
+        MurmurHash,
+        Equal>>
+    linear_hashtable_batch("linear_batch");
```

Comment:
You mean the name? It will be enum in future, right now we hasn't make it clear about all the options and names. Other hashtables will be added one by one

Reply:
yes. what's your concern to change to enum now?

Reply:
This ```enum``` should only be used by cider itself. For other users, they may not use this enum since the ```value``` and ```allocator``` is only compatible with codegen. I can add one enum which is mainly about hashtable's impl and key data type. Like: linear_hash_int, linear_hash_float, chained_hash_int, ck_int8....